addons/account/models/account_journal_dashboard.py

```diff
@@ -196,26 +200,35 @@ def get_bar_graph_datas(self):
 
         return [{'values': data, 'title': graph_title, 'key': graph_key, 'is_sample_data': is_sample_data}]
 
-    def _get_bar_graph_select_query(self):
```

Comment:
This is breaking the stable policy. Some people might have written code that overwrite or add stuff to this method. The method should stay and have the same input,output,parameters as before.

I get the idea of using a default value in the case invoice_date_due is null, that way the returned value of the SQL query will be ordered correctly. There might be another way to fix this that does not break stable policy.
Actually the problem happened due to this PR: https://github.com/odoo/odoo/pull/80627
PSQL was not sending the result in a coherent order so they added an `ORDER BY` to fix the problem. This introduces another problem in case we have no value for the given period because when we order the result, the entry with no value for invoice_date_due will be put at the end resulting in an incoherent order. My guess is that it should first be corrected in v13 and instead of giving a default value in the SQL query like you are doing, it can probably be done by removing the order by and then in python instead of assuming that the result are in the correct place, find the correct place based on the min(invoice_due_date) we received.
So just modifying this part should probably do the trick and won't break the stable policy
```python
for index in range(0, len(query_results)):
            if query_results[index].get('aggr_date') != None:
                is_sample_data = False
                data[index]['value'] = query_results[index].get('total')
```

Reply:
Hello @csnauwaert, I rebased the PR to version 13.0 and corrected my fix. What do you think ?